src/instructlab/configuration.py

```diff
@@ -579,6 +605,11 @@ class Config(BaseModel):
         description="Configuration file structure version.",
         frozen=True,  # don't allow this to be changed anywhere in the code
     )
+    # metadata about the configuration, specifically GPU information
+    metadata: _metadata = Field(
+        default_factory=_metadata,
+        description="Metadata pertaining to the specifics of the system which the Configuration is meant to be applied to.",
```

Comment:
Is it worth calling this something more specific, like `system_metadata` or `sys_metadata`? Do we intend to ever use this for broader, non-system-related metadata?

Reply:
I think leaving it vague is a purposeful choice, in the future I could see this being pretty useful for other config descriptors outside of system hw.